runtime/onert/core/include/ir/train/operation/UntrainableOperation.h

```diff
@@ -30,7 +30,7 @@ namespace onert::ir::train::operation
 // `UntrainableOperation` wraps operations that are not yet supported for training.
 // This class can be removed if all operations are supported for training.
 template <typename OperationType,
-          typename = std::enable_if_t<std::is_base_of<Operation, OperationType>::value>>
+          std::enable_if_t<std::is_base_of_v<Operation, OperationType>, bool> = true>
```

Comment:
According to [cppreference](https://en.cppreference.com/w/cpp/types/enable_if),
> A common mistake is to declare two function templates that differ only in their default template arguments. This does not work because the declarations are treated as redeclarations of the same function template (default template arguments are not accounted for in [function template equivalence](https://en.cppreference.com/w/cpp/language/function_template#Function_template_overloading)).

It's better to use the explicit dummy parameter version (as in the "RIGHT" example). This ensures that the function signatures are distinct and avoids redefinition errors.